pkg/controller/jobframework/setup.go

```diff
@@ -103,6 +109,60 @@ func (m *integrationManager) setupControllers(mgr ctrl.Manager, log logr.Logger,
 	})
 }
 
+func (m *integrationManager) setupControllerAndWebhook(mgr ctrl.Manager, name string, fwkNamePrefix string, cb IntegrationCallbacks, options Options, opts ...Option) error {
+	if err := cb.NewReconciler(
+		mgr.GetClient(),
+		mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, options.ManagerName)),
+		opts...,
+	).SetupWithManager(mgr); err != nil {
+		return fmt.Errorf("%s: %w", fwkNamePrefix, err)
+	}
+	if err := cb.SetupWebhook(mgr, opts...); err != nil {
+		return fmt.Errorf("%s: unable to create webhook: %w", fwkNamePrefix, err)
+	}
+	m.enableIntegration(name)
+	return nil
+}
+
+func waitForAPI(ctx context.Context, mgr ctrl.Manager, checkAPIAvailable fetchRestMapperFunc, log logr.Logger, gvk schema.GroupVersionKind, action func()) {
+	rateLimiter := workqueue.NewItemExponentialFailureRateLimiter(baseBackoffWaitForIntegration, maxBackoffWaitForIntegration)
```

Comment:
what's the rationale for using an exponential backoff instead of just checking every minute?

Reply:
Good question, there could be a scenario where we deploy 2 or more components at the same time, where Kueue was the first one to be deployed. The low base backoff helps in integrating with the new job frameworks quickly, while allowing for a higher-end backoff for any newly added frameworks at any later stage. I'm open to changing it to check on every minute if desired.